### PR TITLE
Try MR normalized input data

### PR DESCRIPTION
**Pull request by ajlee21**
Our collaborator demonstrated that eADAGE model trained on RNA-seq data that was mean-ratio normalized + 0-1 normalized captured a similar number of KEGG pathways compared to eADAGE model trained on array data that was 0-1 normalized. Previously the RNA-seq model was capturing far fewer KEGG pathways compred to the array model.

We used this as inspiration to mean-ratio normalize + 0-1 normalize our recount2 (RNA-seq) data and see if this would remove the RNA-seq only common DEGs.

The main changes to be reviewed are:
1. `normalize_counts.R `: Function that performs the mean-ratio normalization
2. `1a_process_recount2_data_MRnorm.ipynb`: Notebook that processes the recount2 data - applying mean-ratio normalization and 0-1 normalization

The other changes don't need a heavy review as the code is mostly same as was used previously with some additional modifications.

Based on the resulting correlation plot, there is good consistency between SOPHIE common DEGs and common findings generated by manual curation using MR normalization and not using MR normalization. The MR normalization does remove the cluster of genes in the bottom right, but there is more noise in the middle and bottom of the plot compared to the not MR normalized data. Additionally the training curve using the MR normalized data looks fairly flat, which indicates to me that the model is perhaps not learning. I played around with different hyperparams (architecture, batch size, learning rate), but these all didn't change the shape of the curves much. Overall, I'm not sure exactly what is going on here, but maybe further digging should be done when reviews for SOPHIE come back.

Without MR normalization:
image

With MR normalization:
image

### CODE REVIEW

File: human_general_analysis/2a_identify_generic_genes_pathways.py [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+    & (shared_ranking["Percentile (simulated)"] >= percentile_threshold)
+].shape[0]
+# -
+
+print(num_Crow_genes)
+print(num_generic_Crow_genes)
+print(num_generic_SOPHIE_genes)
+print(num_concordant_generic_genes)
+
+p = ss.hypergeom.sf(
```
Comment (jjc2718) [inline human_general_analysis/2a_identify_generic_genes_pathways.py:578]: What is this test showing? Is it that the overlap in common genes between SOPHIE and Crow et al. is very unlikely to be due to chance?

**Review by jjc2718 — APPROVED** [review]
Looks good!

It is interesting that the genes in the bottom right of the hex plot go away. I guess it's promising that you're still correctly identifying common genes (top right), though. Definitely something to think about in the future.

File: generic_expression_patterns_modules/normalize_counts.R [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	# Note the expression data is transposed to gene x sample in order to run Limma
+	expression_data <- t(as.matrix(read.csv(expression_filename, sep="\t", header=TRUE, row.names=1)))
+	metadata <- as.matrix(read.csv(metadata_filename, sep="\t", header=TRUE, row.names=1))
+
+	print("Checking sample ordering...")
+	print(all.equal(colnames(expression_data), rownames(metadata)))
+	#print(colnames(expression_data))
+	dds <- DESeqDataSetFromMatrix(expression_data, colData=metadata, design = ~group)
+	rm(expression_data)
+	dds <- estimateSizeFactors(dds, type="ratio")
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply generic_expression_patterns_modules/normalize_counts.R:15]: No you're right. My comment is not correct. Thanks for catching that!

File: human_general_analysis/2a_identify_generic_genes_pathways.py [hunk 3]
```diff
@@ -0,0 +1,766 @@
...
+        Name for analysis directory. Either "Human" or "Pseudomonas"
+
+    sample_id_colname: str
+        Column header that contains sample id that maps expression data
+        and metadata
+
+    """
+    base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+    if "pseudomonas" in dataset_name.lower():
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply human_general_analysis/2a_identify_generic_genes_pathways.py:149]: No not needed, I can remove that

File: human_general_analysis/2a_identify_generic_genes_pathways.py (hunk 1 above)
Comment (ajlee21) [reply human_general_analysis/2a_identify_generic_genes_pathways.py:578]: Yes, this was a way for us to quantify the correlation between SOPHIE and Crow et al common findings, since the correlation coefficient wasn't very convincing since we're considering all genes in addition to the common ones
